packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -219,6 +232,32 @@ class ImageCache {
     return false;
   }
 
+  /// Updates the least recently used image cache with this image, if it is
+  /// less than the [maximumSizeBytes] of this cache.
+  ///
+  /// Resizes the cache as appropriate to maintain the constraints of
+  /// [maximumSize] and [maximumSizeBytes].
+  void _touch(Object key, _CachedImage image, TimelineTask timelineTask) {
+    assert(timelineTask != null);
+    if (image.sizeBytes != null && image.sizeBytes <= maximumSizeBytes) {
+      _currentSizeBytes += image.sizeBytes;
+      _cache[key] = image;
+      _checkCacheSize(timelineTask);
+    }
+  }
+
+  void _trackLiveImage(Object key, _CachedImage image) {
+    // Avoid adding unnecessary callbacks to the completer.
+    if (_liveImages.containsKey(key)) {
+      assert(identical(_liveImages[key].completer, image.completer));
+      return;
+    }
+    _liveImages[key] = image;
+    image.completer.addOnLastListenerRemovedCallback(() {
```

Comment:
What if nobody ever listens to the stream (and hence this callback never fires)? It appears that we'd never remove the image from the live cache?